components/LanguageToggle.js

```diff
@@ -0,0 +1,149 @@
+          0: "Sunday",
+          1: "Monday",
+          2: "Tuesday",
+          3: "Wednesday",
+          4: "Thursday",
+          5: "Friday",
+          6: "Saturday",
+        },
+        months: {
+          0: "January",
+          1: "February",
+          2: "March",
+          3: "April",
+          4: "May",
+          5: "June",
+          6: "July",
+          7: "August",
+          8: "September",
+          9: "October",
+          10: "November",
+          11: "December",
+        },
+      },
+      ru: {
+        overcast: "ПАСМУРНО",
+        feelsLike: "ОЩУЩАЕТСЯ КАК",
+        wind: "ВЕТЕР",
+        humidity: "ВЛАЖНОСТЬ",
+        latitude: "широта",
+        longitude: "долгота",
+        days: {
+          0: "Воскресенье",
+          1: "Понедельник",
+          2: "Вторник",
+          3: "Среда",
+          4: "Четверг",
+          5: "Пятница",
+          6: "Суббота",
+        },
+        months: {
+          0: "Январь",
+          1: "Февраль",
+          2: "Март",
+          3: "Апрель",
+          4: "Май",
+          5: "Июнь",
+          6: "Июль",
+          7: "Август",
+          8: "Сентябрь",
+          9: "Октябрь",
+          10: "Ноябрь",
+          11: "Декабрь",
+        },
+      },
+    };
+
+    this.setupEventListeners();
+  }
+
+  setupEventListeners() {
+    this.langButton.addEventListener("click", () => {
+      this.switchLanguage();
+    });
+  }
+
+  switchLanguage() {
+    if (this.currentLanguage === "en") {
+      this.currentLanguage = "ru";
+      this.langButton.textContent = "RU";
+    } else {
+      this.currentLanguage = "en";
+      this.langButton.textContent = "EN";
+    }
+
+    this.updateLanguageText();
+  }
+
+  updateLanguageText() {
+    const translations = this.translations[this.currentLanguage];
+
+    this.weatherCondition.textContent = `${translations.overcast}`;
+    this.weatherFeelsLike.textContent = `${translations.feelsLike}: ${this.app.originalTemperatures?.feelsLike}°`;
+
+    this.weatherWind.textContent = `${translations.wind}: ${
+      this.app.originalTemperatures?.wind || "N/A"
+    } m/s`;
+
+    this.weatherHumidity.textContent = `${translations.humidity}: ${
+      this.app.originalTemperatures?.humidity || "N/A"
+    }%`;
+
+    const currentDate = new Date();
+    const dayOfWeek = translations.days[currentDate.getDay()];
+    const dayOfMonth = currentDate.getDate();
+    const month = translations.months[currentDate.getMonth()];
+    this.dateElement.textContent = `${dayOfWeek} ${dayOfMonth} ${month}`;
+
+    if (this.app.location.latitude && this.app.location.longitude) {
+      this.latit.textContent = `${
+        translations.latitude
+      }: ${this.app.location.latitude
+        .toString()
+        .slice(0, 5)
+        .replace(".", "° ")}`;
+      this.langit.textContent = `${
+        translations.longitude
+      }: ${this.app.location.longitude
+        .toString()
+        .slice(0, 5)
+        .replace(".", "° ")}`;
+    }
```

Comment:
Poor readability, first thought was to move into separate method, but then I thought that the better approach would be to split text and values to prevent overlapping. And work only with translation of text.

Temperature values will be changing within temperature-switcher component.

Reply:
https://github.com/VasiliiDubik/Vasiliidubik-weatherForecast/pull/1/commits/56d75da56a2558221c63b3d70234bb9fe3cc063c
Improving code readability by using the formatCoordinates() function from the MapDisplay component.